### PR TITLE
Improve Build and Release Tasks

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
Related to ballerina-platform/ballerina-standard-library/issues/1618

Note: 
- gradle/wrapper/gradle-wrapper.jar is already included in the repository.
- Changes to gradlew and gradlew.bat have already been updated in the main branch.

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -43,19 +43,9 @@ jobs:
...
-          }'
-      - name: Post release PR
+      - name: GitHub Release and Release Sync PR
         env:
           GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
         run: |
-          curl -fsSL https://github.com/github/hub/raw/master/script/get | bash -s 2.14.1
-          bin/hub pull-request --base main -m "[Automated] Sync main after "$VERSION" release"
+          gh release create v$VERSION --title "module-ballerinax-mssql-v$VERSION"
+          gh pr create --title "[Automated] Sync main after $VERSION release" --body "Sync main after $VERSION release"
```
Comment (ThisaruGuruge) [inline .github/workflows/publish-release.yml:51]:
```suggestion
          gh pr create --base main --title "[Automated] Sync main after $VERSION release" --body "Sync main after $VERSION release"
```

**Review by ThisaruGuruge — COMMENTED** [review]
Seems like the gradle wrapper jar is not committed. Shall we check whether there's an issue?

File: .gitignore [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
 # Package Files #
 *.jar
+!gradle/wrapper/gradle-wrapper.jar
```
Comment (ThisaruGuruge) [inline .gitignore:16]: This order has to be changed.

File: .github/workflows/publish-release.yml (hunk 1 above)
Comment (ThisaruGuruge) [reply .github/workflows/publish-release.yml:51]: Shall we add the base branch name here?

File: .gitignore (hunk 2 above)
Comment (kaneeldias) [reply .gitignore]: Fixed in 324dfd1

File: .github/workflows/publish-release.yml (hunk 1 above)
Comment (kaneeldias) [reply .github/workflows/publish-release.yml]: Fixed in 324dfd1

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**